sdk/monitor/Azure.Monitor.Query/TROUBLESHOOTINGGUIDE.md

```diff
@@ -0,0 +1,168 @@
+
+```text
+(BadArgumentError) The request had some invalid properties
+Code: BadArgumentError
+Message: The request had some invalid properties
+Inner error: {
+    "code": "SemanticError",
+    "message": "A semantic error occurred.",
+    "innererror": {
+        "code": "SEM0100",
+        "message": "'take' operator: Failed to resolve table or column expression named 'Appquests'"
+    }
+}
+```
+
+The error message in `innererror` may include the where the Kusto query has an error. You may also refer to the [Kusto Query Language](https://docs.microsoft.com/azure/data-explorer/kusto/query) (KQL) reference docs to learn more about querying logs using KQL.
+
+### Troubleshooting empty log query results
+
+If your Kusto query returns empty or has no logs, validate the following:
+
+- You have the right workspace ID
+- You're setting the correct time interval for the query. Try expanding the time interval for your query to see if that returns any results.
+- If your Kusto query also has a time interval, the query is evaluated for the intersection of the time interval in the query string and the time interval set in the `timeRange` parameter provided the query API. The intersection of these time intervals may not have any logs. To avoid any confusion, it's recommended to remove any time interval in the Kusto query string and use `timeRange` explicitly.
+
+### Troubleshooting server timeouts when executing logs query request
+
+Some complex Kusto queries can take a long time to complete and such queries are aborted by the service if they run for more than 3 minutes. For such scenarios, the query APIs on `LogsQueryClient`, provide options to configure the timeout on the server. The server timeout can be extended up to 10 minutes.
+
+You may see an error as follows:
+
+```text
+Code: GatewayTimeout
+Message: Gateway timeout
+Inner error: {
+    "code": "GatewayTimeout",
+    "message": "Unable to unzip response"
+}
+```
+
+The following code shows an example of setting the server timeout. By setting this server timeout, the Azure Monitor Query library will automatically extend the client timeout to wait for 10 minutes for the server to respond. You don't need to configure your HTTP client to extend the response timeout, as shown in the previous section.
+
+```C# Snippet:QueryLogsWithStatistics
+var client = new LogsQueryClient(new DefaultAzureCredential());
+```
+
+### Troubleshooting partially successful logs query requests
+
+By default, if the execution of a Kusto query resulted in a partially successful response, the Azure Monitor Query client library will throw an exception. The exception indicates to the user that the query wasn't fully successful. The data and the error can be accessed using the `LogsQueryResultStatus` enum and `Error` fields.
+
+```C# Snippet:CreateLogsClient
+var client = new LogsQueryClient(new DefaultAzureCredential());
+```
+
+```csharp
```

Comment:
You'll want to add a code snippet like the following to https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientSamples.cs:

```
        [Test]
        [Explicit]
        public async Task QueryLogsWithPartialSuccess()
        {
            var client = new LogsQueryClient(new DefaultAzureCredential());

            #region Snippet:QueryLogsWithPartialSuccess
            Response<LogsQueryResult> response = await client.QueryWorkspaceAsync(
                "{workspace-id}",
                "{kusto-query-string}",
                new QueryTimeRange(TimeSpan.FromDays(1)),
                new LogsQueryOptions
                {
                    AllowPartialErrors = true
                });
            LogsQueryResult result = response.Value;

            if (result.Status == LogsQueryResultStatus.PartialFailure)
            {
                var errorCode = result.Error.Code;
                var errorMessage = result.Error.Message;

                // code omitted for brevity
            }
            #endregion
        }
```

This code fence should then be changed from `csharp` to `C# Snippet:QueryLogsWithPartialSuccess`. Then you can run the Update-Snippets.ps1 script to inject that code into this snippet placeholder.